SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -385,15 +394,15 @@ void ComponentReader::ApplyArray(Schema_Object* Object, Schema_FieldId FieldId,
 	for (int i = 0; i < Count; i++)
 	{
 		int32 ElementOffset = i * Property->Inner->ElementSize;
-		ApplyProperty(Object, FieldId, *ArrayObjectReferences, i, Property->Inner, ArrayHelper.GetRawPtr(i), ElementOffset, ParentIndex);
+		ApplyProperty(Object, FieldId, *ArrayObjectReferences, i, Property->Inner, ArrayHelper.GetRawPtr(i), ElementOffset, CmdIndex, ParentIndex);
```

Comment:
This looks wrong. Have you tested with arrays of `UObject*`? It looks like when resolving an element of the array, it will use the shadow offset of the array within the replicating object, not the offset of the element within the array.